src/webapp/components/survey/SurveyForm.tsx

```diff
@@ -58,6 +66,11 @@ export const SurveyForm: React.FC<SurveyFormProps> = props => {
     const formClasses = useFormStyles();
     const snackbar = useSnackbar();
     const { api, currentUser } = useAppContext();
+    const [questionnairesSectionsStatus, setQuestionnairesSectionsStatus] = useState<
+        QuestionnairesSectionsStatusType[]
+    >([]);
+    const [questionnaireSections, setQuestionnaireSections] =
+        useState<_.Dictionary<QuestionnaireSection[]>>();
```

Comment:
create a hook for all UI logic - the "how" to render needs to be in custom hooks as per clean architecture.